tests/test_coils.py

```diff
@@ -321,11 +321,30 @@ def test_from_symmetry(self):
         )[0]
         np.testing.assert_allclose(B_true, B_approx, rtol=1e-3, atol=1e-10)
 
+    @pytest.mark.unit
+    def test_from_symmetry_warnings(self):
+        """Test warning in from_symmetry for symmetry plane and self-intersection."""
+        N = 40
+        # test warning after crossing symmetry plane
+        # tilt coils so they cross the symmetry plane
+        coil = FourierPlanarCoil(normal=[1e-4, 1, 3])
+        coils_list_sym = [coil.copy()] + [coil.copy() for i in range(N // 8 - 1)]
+
+        for i, c in enumerate(coils_list_sym[1:]):
+            c.rotate(angle=2 * np.pi / N * (i + 1))
+
+        # test the warning for self-intersecting coils, as two
+        #  of the coils in each field period lie nearly
+        # in the same physical space (intersecting at 2 points) after reflection
+        with pytest.warns(UserWarning) as warninfo:
+            _ = CoilSet.from_symmetry(coils_list_sym, NFP=4, sym=True)
+        assert "nearly intersecting" in str(warninfo[0].message)
```

Comment:
Just a note on the documentation:

This test should really be called `test_is_self_intersecting_warning` since that is where the warning comes from, not the `from_symmetry` method. And you can remove the first comment that says "test warning after crossing symmetry plane" since that doesn't actually raise a warning anymore.